### PR TITLE
issue/70 create department route

### LINKED ISSUE #70: Create Department Route

### PR DESCRIPTION
**Pull request by FoseFx**
Closes #70

### CODE REVIEW

File: routes/departments.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+// @Param      authorization                   header      string                    true    "Bearer: <TOKEN>"
+// @Param      department                      body        CreateDepartmentRequest   true    "Dep. to add"
+// @Success    200                             {object}    SingleDepartmentResponse
+// @Failure    400                             {object}    HTTPErrorResponse
+// @Router     /department                     [put]
+func CreateDepartment(ctx *gin.Context) {
+	log, logCtx := logging.GetRequestLogger(ctx)
+	db := models.GetDB(logCtx)
+
+	//
```
Comment (use-to) [inline routes/departments.go:72]: WHY?
```suggestion
```

File: routes/pagination.go [hunk 2]
```diff
@@ -1,19 +1,58 @@
...
 	"gorm.io/gorm"
+	"rest-api/util"
+	"strconv"
 )
 
 type PaginatedResponse struct {
-	Page      uint
-	PageSize  uint
-	TotalSize uint
+	Page      int
```
Comment (use-to) [inline routes/pagination.go:11]: Is this intended? Is there a negative page number?

File: routes/pagination.go [hunk 3]
```diff
@@ -1,19 +1,58 @@
...
+
+	lastPage := int64(page*pageSize) >= totalSize
+
+	return PaginatedResponse{
+		Page:      page,
+		PageSize:  pageSize,
+		TotalSize: totalSize,
+		LastPage:  lastPage,
+	}, nil
+
```
Comment (use-to) [inline routes/pagination.go:57]:
```suggestion
```

**Review by use-to — CHANGES_REQUESTED** [review]

File: routes/pagination.go (hunk 2 above)
Comment (FoseFx) [reply routes/pagination.go:11]: gorm accepts int, this way we don't need to cast a bunch. openapi docs will only show a generic `integer` anyway, so I see no issue

Comment (use-to) [reply routes/pagination.go:11]: Where is this a gorm related topic?

Comment (FoseFx) [reply routes/pagination.go]: https://github.com/helpwave/rest-api/pull/75/files#diff-9b2ae1c4edb52df1e3a29779c57a265566dd0955af2867052761a7e37b038831R23

File: routes/departments.go [hunk 4]
```diff
@@ -46,3 +47,61 @@ func GetDepartments(ctx *gin.Context) {
...
+// @Param      department                      body        CreateDepartmentRequest   true    "Dep. to add"
+// @Success    200                             {object}    SingleDepartmentResponse
+// @Failure    400                             {object}    HTTPErrorResponse
+// @Router     /department                     [put]
+func CreateDepartment(ctx *gin.Context) {
+	log, logCtx := logging.GetRequestLogger(ctx)
+	db := models.GetDB(logCtx)
+
+	// Validate body
+	//
```
Comment (use-to) [inline routes/departments.go:73]:
```suggestion
```

File: routes/departments.go [hunk 5]
```diff
@@ -46,3 +47,61 @@ func GetDepartments(ctx *gin.Context) {
...
+	body := CreateDepartmentRequest{}
+	if err := ctx.ShouldBindJSON(&body); err != nil {
+		log.Warn().Err(err).Msg("validation failed")
+		SendError(ctx, http.StatusBadRequest, err)
+		return
+	}
+	log.Debug().Str("body", util.Formatted(body)).Send()
+
+	//
```
Comment (use-to) [inline routes/departments.go:82]:
```suggestion
```

File: routes/departments.go [hunk 6]
```diff
@@ -46,3 +47,61 @@ func GetDepartments(ctx *gin.Context) {
...
+	// create model for gORM
+	//
```
Comment (use-to) [inline routes/departments.go:84]:
```suggestion
```

File: routes/departments.go [hunk 7]
```diff
@@ -46,3 +47,61 @@ func GetDepartments(ctx *gin.Context) {
...
+	department := models.Department{
+		DepartmentBase: models.DepartmentBase{Name: body.Name},
+	}
+	log.Debug().Str("model", util.Formatted(department)).Send()
+
+	//
```
Comment (use-to) [inline routes/departments.go:90]:
```suggestion
```

File: routes/departments.go [hunk 8]
```diff
@@ -46,3 +47,61 @@ func GetDepartments(ctx *gin.Context) {
...
+	// add to database
+	//
```
Comment (use-to) [inline routes/departments.go:92]:
```suggestion
```

File: routes/departments.go [hunk 9]
```diff
@@ -46,3 +47,61 @@ func GetDepartments(ctx *gin.Context) {
...
+	res := db.Create(&department)
+	if err := res.Error; err != nil {
+		HandleDBError(ctx, logCtx, err)
+		return
+	}
+
+	//
```
Comment (use-to) [inline routes/departments.go:99]:
```suggestion
```

File: routes/departments.go [hunk 10]
```diff
@@ -46,3 +47,61 @@ func GetDepartments(ctx *gin.Context) {
...
+	// return result
+	//
```
Comment (use-to) [inline routes/departments.go:101]:
```suggestion
```

**Review by use-to — CHANGES_REQUESTED** [review]

**Review by use-to — APPROVED** [review]

File: routes/departments.go [hunk 11]
```diff
@@ -48,6 +49,58 @@ func GetDepartments(ctx *gin.Context) {
...
+// @Param      authorization                   header      string                    true    "Bearer: <TOKEN>"
+// @Param      department                      body        CreateDepartmentRequest   true    "Dep. to add"
+// @Success    200                             {object}    SingleDepartmentResponse
+// @Failure    400                             {object}    HTTPErrorResponse
+// @Router     /departments                    [put]
+func CreateDepartment(ctx *gin.Context) {
+	log, logCtx := logging.GetRequestLogger(ctx)
+	db := models.GetDB(logCtx)
+
+	// Validate body
```
Comment (use-to) [inline routes/departments.go:74]: lowercase maybe

**Pull request merged**